### PR TITLE
extensions: add core22 gnome

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

File: snapcraft/extensions/gnome.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2022 Canonical Ltd
```
Comment (cmatsuoka) [inline snapcraft/extensions/gnome.py:3]: For new files we're using a different copyright line format.

File: snapcraft/extensions/gnome.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+
+from typing import Any, Dict, Optional, Tuple
+
+from overrides import overrides
+
+from .extension import Extension, get_extensions_data_dir
+
+_PLATFORM_SNAP = dict(core22="gnome-42-2204")
+_SDK_SNAP = dict(core22="gnome-42-2204-sdk")
```
Comment (cmatsuoka) [inline snapcraft/extensions/gnome.py:27]: Any special reason for not using literal dictionaries instead of `dict()`?

**Review by cmatsuoka — COMMENTED** [review]
Having a single generic gnome extension looks much nicer than having multiple version-specific ones.

File: snapcraft/extensions/gnome.py (hunk 2 above)
Comment (sergiusens) [reply snapcraft/extensions/gnome.py:27]: copy paste, this can change :-)

File: snapcraft/extensions/gnome.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/extensions/gnome.py:3]: Good catch

File: snapcraft/extensions/gnome.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    At build time it ensures the right build dependencies are setup and for
+    the runtime it ensures the application is run in an environment catered
+    for GNOME applications.
+
+    It configures each application with the following plugs:
+
+    - GTK3 Themes.
+    - Common Icon Themes.
+    - Common Sound Themes.
+    - The GNOME runtime libraries and utilities corresponding to 3.38.
```
Comment (kenvandine) [inline snapcraft/extensions/gnome.py:44]: This should be 42, but you can probably drop the version.

File: snapcraft/extensions/registry.py [hunk 4]
```diff
@@ -16,15 +16,19 @@
...
 
-ExtensionType = Type[Extension]
+if TYPE_CHECKING:
+    from .extension import Extension
 
-_EXTENSIONS: Dict[str, ExtensionType] = {}
+    ExtensionType = Type[Extension]
+
+
+_EXTENSIONS: Dict[str, "ExtensionType"] = {"gnome": GNOME}
```
Comment (kenvandine) [inline snapcraft/extensions/registry.py:31]: Does this mean we won't have a way to use an extension to use a future platform snap, like gnome-44-2204 that is still based on core22?

Comment (sergiusens) [reply snapcraft/extensions/registry.py:31]: this is what we agreed to, remember? If it came to it, it would be a transparent upgrade for users

Comment (kenvandine) [reply snapcraft/extensions/registry.py:31]: I know, and I like the idea in principle but it doesn't give us an easy way to build things that need something newer versions before the next LTS.  Could we add an optional flag of some sort that let's us override the version of the content and build snaps?

Comment (sergiusens) [reply snapcraft/extensions/registry.py:31]: latest commit has a mechanism

File: extensions/desktop/command-chain/Makefile [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/make -f
+
+BIN_DIR              := $(DESTDIR)/snap/command-chain
+
+*:
+	install -D -m755 "$@" "$(BIN_DIR)"/"$@"
+
+install: hooks-configure-fonts desktop-launch run
+
+.PHONY: hooks-configure-fonts desktop-launch run
```
Comment (sergio-costas) [inline extensions/desktop/command-chain/Makefile:10]: Lacks a blank line at the end of the file.

File: snapcraft/extensions/gnome.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+    @functools.cached_property
+    def gnome_snaps(self) -> GNOMESnaps:
+        """Return the GNOME related snaps to use to construct the environment."""
+        base = self.yaml_data["base"]
+        sdk_snap = _SDK_SNAP[base]
+        build_snaps = get_build_snaps(parts_yaml_data=self.yaml_data.get("parts"))
+        matcher = re.compile(r"gnome-\d+-" + _PLATFORM_TRANSLATION[base] + r"-sdk.*")
+        sdk_snap_candidates = [s for s in build_snaps if matcher.match(s)]
+        if sdk_snap_candidates:
```
Comment (sergio-costas) [inline snapcraft/extensions/gnome.py:109]: What happens if the user puts more than one -sdk snap? Is that checked elsewhere?

**Review by sergio-costas — CHANGES_REQUESTED** [review]

**Review by sergio-costas — CHANGES_REQUESTED** [review]
Just some little comments.

**Pull request closed without merging**